docs/cluster_management.md

```diff
@@ -2,45 +2,152 @@
+* GKE [GKE page](https://cloud.google.com/container-engine),
+* Vagrant [Vagrant setup](../docs/getting-started-guides/vagrant.md),
+* GCE [GCE setup](../docs/getting-started-guides/gce.md),
+* CoreOS on Azure [CoreOS on Azure setup](../docs/getting-started-guides/coreos/azure/README.md),
+* CoreOS on AWS [CoreOS on AWS setup](../docs/getting-started-guides/coreos.md),
+* CoreOS on VMWare [CoreOS on VMWare setup](../docs/getting-started-guides/coreos.md).
+
+For more comprehensive and detailed description of cluster setup and configuration process see [From Scratch](../docs/getting-started-guides/scratch.md) guide.
+
+## Difference between user-oriented and ops-oriented tools
+
+Currently Kubernetes has two kinds of tools: developer oriented yet useful for operations people `kubectl` or `cluster/kubectl.sh` that use user specified configuration/contexts. `kubectl` is a stable version of `cluster/kubectl.sh` distributed with the `gcloud` stack.
+
+Cluster configuration/management tools stored in the `cluster` and `hack` folder, which are targeted to operations people, and only people managing clusters should use them. Those tools are mainly `bash` scripts which make managing single cluster easier, but they are currently not designed to allow easy management of multiple clusters from a single machine. If it's necessary it is possible to replicate existing e2e test cluster setup that consists mainly of `hack/ginkgo-e2e.sh` script and special test configs for specific cloud providers e.g. `cluster/gce/config-test.sh`. This means that in general scripts from `cluster` directory find a running cluster by themselves, and you don't need to explicitly specify the cluster you want to modify, as those tools assume that there's the cluster running.
+
 ## Upgrading a cluster
 
-The `cluster/kube-push.sh` script will do a rudimentary update; it is a 1.0 roadmap item to have a robust live cluster update system.
+GKE handles cluster upgrades for you, so you don't need to worry about. If you already have a cluster running in some other environment and want to upgrade Kubernetes version running on it, you can use the `cluster/kube-push.sh` script. It works out of the box if your cluster runs on GCE and most of other providers. The upgrade is in-place, meaning that all user pods are kept alive throught this process.
 
-## Updgrading to a different API version
+`kube-push.sh` has a few functionalities. It can either be used to upgrade all Kubernetes components to the currently built binaries (default option), or to do limited upgrade of either the master machine or a single node. Instead of upgrading to the head, you can also specify the version which you want to roll out.
 
-There is a sequence of steps to upgrade to a new API version.
+If you made some changes to the release you're running, or downloaded the head repository, and want to upgrade all components, you can run following commands from the `kubernetes` directory:
+```
+make release
```

Comment:
IIUC it's because you want to have an 'admin-not-dev' guide, right? I'll comply, but I'm not really sure I agree with very sharp distinction.